src/components/books/books.js

```diff
@@ -1,53 +1,87 @@
-/* eslint-disable react/jsx-wrap-multilines */
-import { useDispatch, useSelector } from 'react-redux';
-import { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useEffect, useState } from 'react';
 import Book from './book/book';
+import { addBook, API } from '../../redux/books/books';
 import './books.css';
-import { addBook } from '../../redux/books/books';
 
 const Books = () => {
   const [book, setBook] = useState('');
-  const [author, setAuthor] = useState('');
+  const [category, setCategory] = useState('');
   const data = useSelector((state) => state.booksReducer);
-
   const dispatch = useDispatch();
-  const submitBookToStore = () => {
-    if (book.length > 0 && author.length > 0) {
+
+  useEffect(async () => {
+    const response = await fetch(`${API}`);
+    const Books = await response.json();
+    const booksID = Object.keys(Books);
+    const books = booksID.map((key) => {
+      const book = Books[key][0];
+      book.id = key;
+      return book;
+    });
+    books.forEach((book) => {
+      dispatch(addBook(book));
+    });
+  }, []);
```

Comment:
[OPTIONAL] I highly recommend that when using redux, all logic that modifies the state of the reducer goes into the slice reducers. In this case, using thunks. This was a requirement, but since the last reviewer did not notice I will mark it as optional, but again, I highly recommend study this, [you can find more information here.](https://redux.js.org/tutorials/fundamentals/part-6-async-logic#using-middleware-to-enable-async-logic)

> The redux code of the application should be refactored using *middleware* to make async requests to the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739).

I also notice there is a small bug on the app, I highly recommend you fix it. When this component mounts, useEffect is always being run and books get added to the store. Resulting in the following bug:

![Book-rerenders](https://user-images.githubusercontent.com/79114233/145682347-7ff0be35-5bce-46a9-b255-b960dd61f2d1.gif)

Common fixes for this are checking if the store is empty before doing more fetching. This will also improve the performance of the App.